CI breaking

**Describe the bug**
CI is now breaking likely due to some changes in mypy version.
```
tests/extension/decimal/array.py:83: error: No overload variant of "__setitem__" of "ndarray" matches argument types "int", "Decimal"  [call-overload]
tests/extension/decimal/array.py:83: note: Possible overload variants:
tests/extension/decimal/array.py:83: note:     def __setitem__(self, str | list[str], Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
tests/extension/decimal/array.py:83: note:     def __setitem__(self, SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | tuple[SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | None, ...] | None, Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
tests/extension/decimal/array.py:86: error: No overload variant of "__setitem__" of "ndarray" matches argument types "int", "Decimal"  [call-overload]
tests/extension/decimal/array.py:86: note: Possible overload variants:
tests/extension/decimal/array.py:86: note:     def __setitem__(self, str | list[str], Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
tests/extension/decimal/array.py:86: note:     def __setitem__(self, SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | tuple[SupportsIndex | slice | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | None, ...] | None, Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes], /) -> None
tests/test_scalars.py:582: error: Expression is of type "timedelta", not "Timedelta"  [assert-type]
tests/test_scalars.py:641: error: Expression is of type "timedelta", not "Timedelta"  [assert-type]
Found 4 errors in 2 files (checked [22](https://github.com/pandas-dev/pandas-stubs/actions/runs/12307927720/job/34352450935?pr=1070#step:4:23)8 source files)
```

**To Reproduce**
Run `poetry run poe test_all`.

**Please complete the following information:**
 - OS: [e.g. Windows, Linux, MacOS] All
 - OS Version [e.g. 22] latest
 - python version 3.10 to 3.12
 - version of type checker
 - version of installed `pandas-stubs`


**Additional context**
Add any other context about the problem here.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ packages = [{ "include" = "pandas-stubs" }]
 [tool.poetry.dependencies]
 python = ">=3.10"
 types-pytz = ">= 2022.1.1"
-numpy = ">= 1.23.5"
+numpy = ">= 1.23.5,< 2.2.0" # wait till numpy 2.2.1 to clear
 
 [tool.poetry.group.dev.dependencies]
 mypy = "1.13.0"

